zoom_monitor.py

```diff
@@ -148,48 +152,39 @@ def _open_participants_list(self):
                 time.sleep(1)
                 continue  # Go to the next attempt
 
-            selected = self._is_participants_button_selected()
+            # Sometimes, the button is hidden off the bottom of the window,
+            # but moving the mouse to it will make it visible again. This
+            # tends to happen after someone stops sharing their screen.
+            ActionChains(self._driver).move_to_element(button).perform()
             try:
-                # Clicking on the participants list only selects the button.
-                # As a small clue: if a human clicks on it, the mouse-down
-                # selects the button while the mouse-up opens the participants
-                # list. Double-clicking on it seems to work okay (maybe the
-                # second click implicitly creates a mouse-up on the first one).
-                # If the button is already selected, only one click is needed.
                 button.click()
-                if not selected:
-                    button.click()  # Channeling our inner grandma
-            except ElementClickInterceptedException:
-                self._logger.debug("DOM isn't set up; wait and try again")
+            except (ElementClickInterceptedException,
+                    ElementNotInteractableException) as e:
+                self._logger.debug(f"DOM isn't set up ({e}); try again soon.")
                 time.sleep(1)
                 continue  # Go to the next attempt
-
             self._logger.debug("participants list clicked")
-            # Now that we've clicked the participants list without raising
-            # an exception, wait until it shows up.
-            self._wait_for_element(
-                By.CLASS_NAME, "participants-wrapper__inner")
+
+            try:
+                # Now that we've clicked the participants list without raising
+                # an exception, wait until it shows up. If it doesn't show up
+                # yet, it might be that we've highlighted the button but
+                # haven't properly clicked it, and the next iteration's attempt
+                # will succeed.
+                self._wait_for_element(
+                    By.CLASS_NAME, "participants-wrapper__inner")
+            except TimeoutException:
+                self._logger.debug("timed out waiting for participants list,"
+                                   "will try clicking again soon.")
+                time.sleep(1)
```

Comment:
It looks like `self._wait_for_element` includes a `5s` timeout; do we need this sleep if that's the case?

Reply:
Great point! I'll remove it.